google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -556,6 +567,12 @@ def execute
               rescue Google::Apis::Error => e
                 raise e unless retry? e.body, current_retries
 
+                if @logger
+                  msg = ["Attempting retry #{current_retries + 1} because of retryable error: #{e.inspect}"]
```

Comment:
one of the issues with the old error was that it included a fragment of the payload, which might be sensitive, in the logs.

```
irb(main):003:0> begin; JSON.parse('{"hello": {world}');  rescue; puts $!.inspect; end
#<JSON::ParserError: 783: unexpected token at '{"hello": {world}'>
```

maybe do `#class` instead of `#inspect` ?

```
irb(main):004:0> begin; JSON.parse('{"hello": {world}');  rescue; puts $!.class; end
JSON::ParserError
```

Reply:
Yes, of course. What do you think about two logging statements: The current `warn` statement would be changed to safely log just `#class`, and a `debug` statement could be added to log the `#inspect` output in controlled environments? Or is there no value in seeing the payload?

Reply:
I'll change the logged output now to just the `#class` info. If anyone feels strongly that more detail is needed, we can add that later in a separate debug-level statement.

Reply:
@shollyman requested in #5180:

> Any chance we have a reasonably clean way to log the full response from the backend when we encounter this kind of parse error? May help the reporter get more info, I suspect reproducing this one without more info is challenging.

Reply:
I updated this PR to log just the `#class` info at `WARN` level, and the full `#inspect` output at `DEBUG` level. This is now described in `LOGGING.md`.